src/server/upload-and-scan/controller.js

```diff
@@ -82,6 +87,14 @@ const uploadController = {
       uploadDetails.pending = new Date().toISOString()
       await request.redis.storeUploadDetails(uploadId, uploadDetails)
 
+      // If a file was rejected during the upload trigger processScanComplete
+      // This will ensure the overall status gets updated.
+      for (const s of fileStatuses) {
+        if (s.status === fileStatus.rejected) {
+          await processScanComplete(request.server, uploadId, s.fileId)
+        }
```

Comment:
If there are multiple rejected files in `fileStatuses` will this call `processScanComplete()` multiple times?

Can I suggest you do a 

```
if (fileStatuses.some((file) => file.status === fileStatus.rejected)) {
 await processScanComplete(request.server, uploadId, s.fileId)
}
```

Reply:
as i understand it, the processScanComplete function takes a file id so needs to be called once per file